src/main/java/com/puppycrawl/tools/checkstyle/checks/coding/RequireThisCheck.java

```diff
@@ -89,53 +89,144 @@
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;/&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(arg);         // OK, no overlap
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
+ *   }
+ * }
+ * </pre>
  * <p>
  * To configure to check the {@code this} qualifier for fields only:
  * </p>
+ * <p>(with validateOnlyOverlapping set to true)</p>
  * <pre>
  * &lt;module name=&quot;RequireThis&quot;&gt;
  *   &lt;property name=&quot;checkMethods&quot; value=&quot;false&quot;/&gt;
+ *   &lt;property name=&quot;validateOnlyOverlapping&quot; value=&quot;true&quot;/&gt;
  * &lt;/module&gt;
  * </pre>
+ * <p>Example:</p>
+ * <pre>
+ * public class Test {
+ *   private int a;
+ *   private int b;
+ *   private int c;
+ *
+ *   public Test(int a) {
+ *     // overlapping by constructor argument
+ *     this.a = a;       // OK, this keyword used
+ *     b = 0;            // OK, no overlap
+ *     foo(arg);         // OK, no validation for methods
+ *   }
+ *
+ *   public void foo(int c) {
+ *     // overlapping by method argument
+ *     c = c;            // violation, reference to instance variable "c" requires "this"
```

Comment:
if config is different from default, thre should difference in traiing comments (more violations or less violations), please udpate example block to make it.

Reply:
Hey, I tried making each example different, but the problem is that instance methods never actually require `this` keyword because they cannot be overlapped in Java. I searched for examples on StackOverflow as well but could not find a situation where `this` keyword must be used with a method. 
That is why there is no violation when checking only for methods (with validateOnlyOverlapping = true).
Please guide on how to proceed.

Reply:
`this` is never required, compiler will do his job in any case.
it is matter of style and consistency that user want to have over code.
`checkMethods` was added in version 3.4 it is like beginning of this century, I am not sure why somebody care about this at that time.

in scope of this issue we just should show how Check works now. So please create some line like "test()" and lets see where we place violation.

Reply:
There is already a method call `foo()` on line 133 which is not raising violations, and I cannot think of any case where it would. It may be that I am not able to understand what you meant by your last statement. Sorry for the delay.

Reply:
To show the difference between default check and configured check, I could change the order of the examples so that user would understand the difference in violations and code.

Reply:
sure , please do , but default config is first.

Reply:
Done, I have changed order of examples, hopefully it is easier to understand now.